src/App/Pages/Vault/AddEditPageViewModel.cs

```diff
@@ -449,6 +461,10 @@ public async Task<bool> SubmitAsync()
                 }
                 else
                 {
+                    if(CloneMode && ViewPage != null)
+                    {
+                        ViewPage.ViewModel.CipherId = this.Cipher.Id;
```

Comment:
Let's make a public method on the ViewPage that lets you pass in the new id. Then you can set the ViewModel and the private `_cipherId` member as well.